cmd/platform/server_test.go

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package main
+
+import (
+	"os"
+	"syscall"
+	"testing"
+
+	"github.com/mattermost/mattermost-server/jobs"
+	"github.com/mattermost/mattermost-server/utils"
+	"github.com/stretchr/testify/require"
+)
+
+type ServerTestHelper struct {
+	configPath         string
+	disableConfigWatch bool
+	signalChannel      chan os.Signal
+	originalInterval   int
+}
+
+func SetupServerTest() *ServerTestHelper {
+	// Build a channel that will be used by the server to receive system signals…
+	signalChannel := make(chan os.Signal, 1)
+	// …and sent it immediately a SIGINT value.
+	// This will make the server loop stop as soon as it started successfully.
+	signalChannel <- syscall.SIGINT
+
+	// Let jobs poll for termination every 0.2s (instead of every 15s by default)
+	// Otherwise we would have to wait the whole polling duration before the test
+	// terminates.
+	originalInterval := jobs.DEFAULT_WATCHER_POLLING_INTERVAL
+	jobs.DEFAULT_WATCHER_POLLING_INTERVAL = 200
+
+	th := &ServerTestHelper{
+		configPath:         utils.FindConfigFile("config.json"),
+		disableConfigWatch: true,
+		signalChannel:      signalChannel,
+		originalInterval:   originalInterval,
+	}
+	return th
+}
+
+func (th *ServerTestHelper) TearDownServerTest() {
+	jobs.DEFAULT_WATCHER_POLLING_INTERVAL = th.originalInterval
+}
```

Comment:
Note to the reviewer: creating a TestHelper struct (with Setup and TearDown methods) is the best way I found to cleanup the jobs polling interval after the tests.

Is it a good location for this TestHelper to be right in the test file? Or is there a better approach?

Reply:
I think this is perfectly fine.